src/NexusMods.App.UI/Pages/Downloads/InProgressViewModel.cs

```diff
@@ -127,37 +142,103 @@ public class InProgressViewModel : APageViewModel<IInProgressViewModel>, IInProg
 
-        TaskSourceChangeSet = downloadService.Downloads
-            .ToObservableChangeSet(x => x.PersistentState.Id)
+        var tasksChangeSet = _downloadService.Downloads
+            .ToObservableChangeSet(x => x.PersistentState.Id);
+        
+        InProgressTaskChangeSet = tasksChangeSet
             .Transform(x =>
                 {
                     var vm = new DownloadTaskViewModel(x);
                     vm.Activator.Activate();
                     return (IDownloadTaskViewModel)vm;
                 }
             )
-            .FilterOnObservable((item, key) => item.WhenAnyValue(v => v.Status)
-                .Select(s => s != DownloadTaskStatus.Cancelled && s != DownloadTaskStatus.Completed))
+            .FilterOnObservable((item, key) =>
+                {
+                    return item.WhenAnyValue(v => v.Status)
+                        .Select(s => s != DownloadTaskStatus.Cancelled && s != DownloadTaskStatus.Completed);
+                }
+            )
+            .AutoRefreshOnObservable(task =>
+                {
+                    return task.WhenAnyValue(x => x.Status);
+                }
+            )
             .DisposeMany()
             .OnUI();
+            
+
+        CompletedTaskChangeSet = tasksChangeSet
+            .Transform(x =>
+                {
+                    var vm = new DownloadTaskViewModel(x);
+                    vm.HideCommand = ReactiveCommand.CreateFromTask(async () => await HideTasks(true, [vm]));
+                    vm.ViewInLibraryCommand = ReactiveCommand.Create<NavigationInformation>((navInfo) =>
+                    {
+                        var controller = GetWorkspaceController();
+                        var workspaces = controller
+                            .AllWorkspaces
+                            .Where(w => w.Context is LoadoutContext loadoutContext
+                                        && conn.Db.Get<Loadout.Model>(loadoutContext.LoadoutId.Value).Installation.Game.Domain.Equals(vm.Game)
+                            )
+                            .Select(w => (w.Id, Context: (LoadoutContext)w.Context)).ToArray();
+                        
+                        if (workspaces.Length == 0)
+                            return;
+                        
+                        var workspace = workspaces[0];
+
+                        var pageData = new PageData
+                        {
+                            FactoryId = FileOriginsPageFactory.StaticId,
+                            Context = new FileOriginsPageContext { LoadoutId = workspace.Context.LoadoutId },
+                        };
+                        var behavior = GetWorkspaceController().GetOpenPageBehavior(pageData, navInfo, Optional<PageIdBundle>.None);
+                        
+                        controller.OpenPage(workspace.Id, pageData, behavior);
+                        controller.ChangeActiveWorkspace(workspace.Id);
+                    });
```

Comment:
Why did you decide to use this approach instead of having empty commands where you listen to command execution to run this logic?

Reply:
Because I needed the task data and the Navigation info data for the logic and this approach made it much simpler to have that data available.
It also made async easier to deal with since it all goes into the `CreateFromTask`.